### PR TITLE
feat/158 - Angular15 admin standalone components

### PR DESCRIPTION
**Pull request by ianliuwk1019**
Upgrade 'admin' side to Angular 15 standalone components.

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
 - [api](https://fom-357.apps.silver.devops.gov.bc.ca/api)
 - [admin](https://fom-357.apps.silver.devops.gov.bc.ca/admin)
 - [public](https://fom-357.apps.silver.devops.gov.bc.ca/public)

Once merged, code will be promoted and handed off to following workflow run.
 - [Main Merge Workflow](https://github.com/bcgov/nr-fom/actions/workflows/merge-main.yml)

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
 - [api](https://fom-357.apps.silver.devops.gov.bc.ca/api)
 - [admin](https://fom-357.apps.silver.devops.gov.bc.ca/admin)
 - [public](https://fom-357.apps.silver.devops.gov.bc.ca/public)

Once merged, code will be promoted and handed off to following workflow run.
 - [Main Merge Workflow](https://github.com/bcgov/nr-fom/actions/workflows/merge-main.yml)

### CODE REVIEW

File: admin/src/app/app.component.ts [hunk 1]
```diff
@@ -1,11 +1,23 @@
-import {Component, OnInit} from '@angular/core';
-import {StateService} from '../core/services/state.service';
-import {Observable} from 'rxjs';
+import { AsyncPipe, NgIf } from '@angular/common';
+import { Component, OnInit } from '@angular/core';
+import { RouterOutlet } from '@angular/router';
+import { Observable } from 'rxjs';
+import { StateService } from '../core/services/state.service';
```
Comment (basilv) [inline admin/src/app/app.component.ts]: Can we use imports via @ instead of these relative paths?

File: admin/src/app/foms/fom.resolvers.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { inject } from '@angular/core';
+import { ActivatedRouteSnapshot, ResolveFn, RouterStateSnapshot } from '@angular/router';
+import { ProjectMetricsResponse, ProjectResponse, ProjectService, SpatialFeaturePublicResponse, SpatialFeatureService } from '@api-client';
+import { ConstantUtils } from '../../core/utils/constants/constantUtils';
```
Comment (basilv) [inline admin/src/app/foms/fom.resolvers.ts]: Another example of relative import that could be changed.

File: admin/src/app/header/header.component.ts [hunk 3]
```diff
@@ -1,22 +1,26 @@
 import { animate, state, style, transition, trigger } from '@angular/animations';
+import { NgIf } from '@angular/common';
 import { Component, OnInit } from '@angular/core';
-import { Router } from '@angular/router';
+import { Router, RouterLink } from '@angular/router';
+import { NgbNavbar } from '@ng-bootstrap/ng-bootstrap';
 import { User } from "@utility/security/user";
 import { ConfigService } from '@utility/services/config.service';
 import { CognitoService } from "../../core/services/cognito.service";
```
Comment (basilv) [inline admin/src/app/header/header.component.ts]: Relative path import.

File: admin/src/app/search/search.component.ts [hunk 4]
```diff
@@ -12,9 +15,22 @@ import { ModalService } from '../../core/services/modal.service';
 import { StateService } from '../../core/services/state.service';
```
Comment (basilv) [inline admin/src/app/search/search.component.ts]: Relative path import.

File: admin/src/core/components/button/button.component.ts [hunk 5]
```diff
@@ -1,9 +1,10 @@
-import { Component, Input, OnInit } from '@angular/core';
+import { Component, Input } from '@angular/core';
 import { StateService } from '../../../core/services/state.service';
```
Comment (basilv) [inline admin/src/core/components/button/button.component.ts]: pre-existing relative path import.

File: admin/src/app/app.component.ts (hunk 1 above)
Comment (ianliuwk1019) [reply admin/src/app/app.component.ts]: Should be good now, thanks Basil.

**Review by basilv — APPROVED** [review]
Looks good.

**Pull request merged**